### PR TITLE
Fix/mutate promise error

### PR DESCRIPTION
**Pull request by mbirkegaard**
All credit to @Emilios1995. I'm just the one putting up the PR. 

This fixes the type of the promise returned by `mutate`.

On the ts side, `useMutation` returns a tuple with this definition:
```
export type MutationTuple<TData, TVariables> = [
  (
    options?: MutationFunctionOptions<TData, TVariables>
  ) => Promise<ExecutionResult<TData>>,
  MutationResult<TData>
];
```
where 
```
export interface ExecutionResult<T = Record<string, any>> {
  data?: T;
  extensions?: Record<string, any>;
  errors?: GraphQLError[];
}
```
The previous reason types assumed that `mutate` returned `Promise<MutationResult<TData>>` instead of `Promise<ExecutionResult<TData>>`

This is a breaking change. Particularly annoying after #89.

### CODE REVIEW

**Comment by fakenickels** [comment]
Will review soon

File: src/ApolloHooksMutation.re [hunk 1]
```diff
@@ -140,7 +146,7 @@ let useMutation:
                (
                  switch (
                    Js.Nullable.toOption(jsResult##data),
-                   Js.Nullable.toOption(jsResult##error),
+                   Js.Nullable.toOption(jsResult##errors),
                  ) {
                  | (_, Some(error)) => (Error(error): result('data))
```
Comment (Emilios1995) [inline src/ApolloHooksMutation.re]: @mbirkegaard I think we should add a guard to this pattern branch `when Js.Array.length(error) > 0`.

Comment (mbirkegaard) [reply src/ApolloHooksMutation.re]: Yep.

**Comment by mbirkegaard** [comment]
@baransu When it comes time to release this, can you do a release candidate first? We're starting to migrate a lot of users to our Reason platform so there's a chance that we'll discover more bugs. It would be annoying to keep making braking changes as we fix them.

**Comment by fakenickels** [comment]
Of course! We have been churning a lot of the last changes

**Comment by fakenickels** [comment]
Sorry about the delay to merge, we have been a bit busy these past few weeks

**Comment by mbirkegaard** [comment]
> Sorry about the delay to merge, we have been a bit busy these past few weeks

No worries _at all!_ It's OSS! Noone is paying, so noone is owed anything ;)

**Comment by fakenickels** [comment]
🙏 
@mbirkegaard are you on ReasonML Discord server? If so what's your handle there?

**Comment by mbirkegaard** [comment]
I'm on there, though I only pop in occasionally. My handle is FreddieFreeloader

**Pull request merged**